server/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -139,7 +139,12 @@ public enum Property {
         /**
          * Indicates an index-level setting that is privately managed. Such a setting can not even be set on index creation.
          */
-        PrivateIndex
+        PrivateIndex,
+
+        /**
+         * Indicates that this index-level setting was deprecated and removed in the current major
+         */
+        DeprecatedAndRemovedInCurrentMajor
```

Comment:
Likewise here, let's mention the specific versions in the symbol name and trigger a compile error in v9. Also this property only makes sense for index settings. How about something like this:

```suggestion
        /**
         * Indicates that this index-level setting was deprecated in {@link Version#V_7_17_0) and removed in {@link Version#V_8_0_0}.
         */
        IndexSettingDeprecatedInV7AndRemovedInV8
```

Reply:
I wonder if we have to mention Deprecation in the property name.
howe about `Property.ToBeRemovedInV8IndexSetting`? or `Property.IndexSettingToBeRemovedInV8`?

Reply:
I could live with that too, maybe even just `IndexSettingRemovedInV8`, although I wonder if it's helpful to be explicit about the property's effects in its name. We must mention `Version#V_7_17_0` in the javadoc to ensure this property is removed in v9.